### PR TITLE
[build] Add cmake option to enable/disable Java components build

### PR DESCRIPTION
**Pull request by brnorris03**
Currently, if JNI is detected, the build always attempts to compile the Java runtime and tests. This patch adds a cmake option `ONNX_MLIR_ENABLE_JAVA` to enable (by default) or disable the Java components build.

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
@jenkins-droid test this please

**Comment by AlexandreEichenberger** [comment]
(merged with the latest to start the CIs).

@gongsu832 do you mind looking into this feature, and see if it interact with how our CI works when testing for java, Thanks

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,34 +88,36 @@ endif()
...
-if (Java_Development_FOUND AND JNI_FOUND)
-  set(ONNX_MLIR_ENABLE_JNI TRUE)
-  include(UseJava)
+  if (Java_Development_FOUND AND JNI_FOUND)
+    set(ONNX_MLIR_ENABLE_JNI TRUE)
+    include(UseJava)
+  endif()
 else()
   set(ONNX_MLIR_ENABLE_JNI FALSE)
   message(WARNING "Java Development component or JNI not found, JNI targets will not work")
```
Comment (gongsu832) [inline CMakeLists.txt:123]: This message is for the case when `ONNX_MLIR_ENABLE_JAVA` is ON but Java Development component or JNI is not found, not when `ONNX_MLIR_ENABLE_JAVA` is OFF.

Comment (brnorris03) [reply CMakeLists.txt]: Oops, sorry about that! I restored the `"Java Development component or JNI not found, JNI targets will not work"` message to its proper place and added a different message (`"Java explicitly disabled with the ONNX_MLIR_ENABLE_JAVA option, JNI targets will not work"`) when the new option is used to disable Java explicitly.

**Comment by AlexandreEichenberger** [comment]
@gongsu832 I assumed that you approved the PR? You left your `LGTM` as a comment and not approval.

**Comment by gongsu832** [comment]
Yes. Sorry forgot to click the approve button.

**Pull request merged**